contract-sdk/src/env.rs

```diff
@@ -23,4 +24,28 @@ pub trait Env {
 pub trait Crypto {
     /// ECDSA public key recovery function.
     fn ecdsa_recover(&self, input: &[u8]) -> [u8; 65];
+
+    /// Verify a message signature.
+    fn signature_verify(
+        kind: SignatureKind,
+        key: &[u8],
+        context: &[u8],
+        message: &[u8],
+        signature: &[u8],
+    ) -> bool;
+
+    /// Verify a batch of messages and signatures using the same public key.
+    ///
+    /// If the lengths of the contexts, messages and signatures slices don't
+    /// match or if there is some other error with the inputs, an Err(()) will
+    /// be returned. Otherwise, the booleans in the returned vector are the
+    /// verification results for each triplet of inputs.
+    #[allow(clippy::result_unit_err)]
+    fn signature_verify_batch(
```

Comment:
Whowa, that's an odd definition of batch verification.  Why the fixed public key.  Also AFAIK, we do not have a consensus-safe rust side ed25519 batch verify that follows our verification semantics...